packages/website/ts/blockchain.ts

```diff
@@ -611,6 +576,45 @@ export class Blockchain {
 
         this._dispatcher.updateBlockchainIsLoaded(true);
     }
+    private async _getInjectedProviderIfExistsAsync(): Promise<InjectedProvider | undefined> {
+        if (!_.isUndefined(this._injectedProviderIfExists)) {
+            return this._injectedProviderIfExists;
+        }
+        let injectedProviderIfExists = (window as any).ethereum;
+        if (!_.isUndefined(injectedProviderIfExists)) {
+            if (!_.isUndefined(injectedProviderIfExists.enable)) {
```

Comment:
if `injectedProvider` is not undefined but it does not contain `enable`, are we intentionally setting `this._injectedProviderIfExists = injectedProviderIfExists;` at the end of this function?

Reply:
I get that the fall-through is intentional, but don't get why there are two different if-statements here. Anyway I don't think it makes a difference.